server/src/main/java/timely/api/model/Metric.java

```diff
@@ -129,7 +144,10 @@ public Mutation toMutation() {
 
     public static Metric parse(Key k, Value v) {
         ComparablePair<String, Long> row = rowCoder.decode(k.getRow().getBytes());
-        Double value = valueCoder.decode(v.get());
+        READ_BUFFER.get().clear();
+        READ_BUFFER.get().put(v.get());
```

Comment:
I believe you need to account for the fact that the byte buffer returned by a Value may be larger than the actual data in it. I.e., I think you need ```READ_BUFFER.get().put(v.get(), 0, v.getSize())```.